File: frontend/src/app/responses/recommendation/recommendation-response.ts
import { CourseRecommendationItemResponse } from './course-recommendation-item-response';
import { BoCauHoiRecommendationItemResponse } from './bo-cau-hoi-recommendation-item-response';

export interface RecommendationResponse {
  courses: CourseRecommendationItemResponse[];
  bo_cau_hoi: BoCauHoiRecommendationItemResponse[];
}

File: backend/src/main/java/com/app/backend/responses/recommendation/RecommendationResponse.java
package com.app.backend.responses.recommendation;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RecommendationResponse {

    @JsonProperty("courses")
    private List<CourseRecommendationItemResponse> courses;

    @JsonProperty("bo_cau_hoi")
    private List<BoCauHoiRecommendationItemResponse> boCauHoi;
}


